### PR TITLE
Rename project to "Tekton Pipelines"

### PR DESCRIPTION
**Pull request by bobcatfish**
Instead of calling this "Pipeline CRD" or "Knative Pipelines", we'd like
to call this project "Tekton Pipelines". Before our first release, 0.1,
which we are intentionally decoupling from the Knative release process,
we want to change the api groups and other resources to reflect the new name.

The release scripts are hard-coded to use `knative` as the name, so we
will have to work around that for now.

### CODE REVIEW

**Comment by abayer** [comment]
Woo!
/meow space

**Comment by abayer** [comment]
@bobcatfish Looks like you need to `update-codegen.sh`. And rebase, but hey. =)

**Comment by afrittoli** [comment]
Do we really want to go through with renaming to a name that is protected by copyright already (https://www.tekton.com/)? The renaming itself is pretty disruptive, it's ok at this stage of the project but another rename would be problematic later on.

File: hack/release.md [hunk 1]
```diff
@@ -50,36 +50,13 @@ Examples:
 
 _Note: only Knative admins can create versioned releases._
```
Comment (pivotal-nader-ziada) [inline hack/release.md]: Tekton admins?

Comment (bobcatfish) [reply hack/release.md]: hm it probably is still actually Knative admins right now, I'm guessing that's how the permissions are set up 🤔

**Comment by bobcatfish** [comment]
> Do we really want to go through with renaming to a name that is protected by copyright already (https://www.tekton.com/)?

That's a good point @afrittoli - since I am not a lawyer, I can't comment on the specific legalities here, 
we did run this by the google lawyers before going ahead.

> The renaming itself is pretty disruptive, it's ok at this stage of the project but another rename would be problematic later on.

Apologies for the disruption 😭😭😭

**Comment by hrishin** [comment]
Are we going to move/rename this project to the different repo?

**Pull request merged**